firedrake/ffc_interface.py

```diff
@@ -265,7 +260,9 @@ def compile_form(form, name, parameters=None, inverse=False):
     fd = compute_form_data(form)
 
     # If there is no mixed element involved, return the kernels FFC produces
-    if all(isinstance(e, (FiniteElement, VectorElement)) for e in fd.unique_sub_elements):
+    # Note: using type rather than isinstance because UFL's VectorElement,
```

Comment:
Sounds like the design of our (UFL's?) finite element type hierarchy violates the Liskov substitution principle.
Is it because the representation (memory layout?) of VectorElement / MixedElement is different in Firedrake and in FEniCS/DOLFIN?